trulens_eval/trulens_eval/tests/benchmark_frameworks/eval_as_recommendation.py

```diff
@@ -0,0 +1,104 @@
+import pandas as pd
+import numpy as np
+from sklearn.metrics import ndcg_score
+from typing import List
+import time
+import random
+
+import logging  
+log = logging.getLogger(__name__)
+
+
+
+"""score passages with feedback function, retrying if feedback function fails.
+    Args: df: dataframe with columns 'query_id', 'query', 'passage', 'is_selected'
+            feedback_func: function that takes query and passage as input and returns a score
+            backoff_time: time to wait between retries
+            n: number of samples to estimate conditional probabilities of feedback_func's scores
+"""
+def score_passages(df, feedback_func_name, feedback_func, backoff_time=0.5, n=5, temperature=0.0):
```

Comment:
can this me a bit more generalized to take the name of a feedbak function and a list of trulens record_ids or similar rather than constructing a new df?

Reply:
I guess I didn't use the best naming convention here - the passed-in dataframe is usually comprised of some human annotated/curated datasets (i.e. for context relevance case it was constructed from MS MARCO and thus has the "is_selected" field) 

I 100% agree this method should definitely be more generic but off top of my head I'm not sure if taking in trulens' record_ids woud work when some of the cases groundtruths are not available for the downstream metrics calculation of ECE, NDCG, etc.

Happy to take it on a separate PR if you're okay with this piece of ad-hoc code going into the main branch